### DISCUSSION TITLE
Listing all external dependencies name, version and licence(s) for target

### DISCUSSION THREAD

**Discussion by aallrd** [bazelbuild/bazel#20585] [Q&A]
Hello,

I have to list all the external dependencies of a `bazel` built project ([envoy](https://github.com/envoyproxy/envoy)), with their name, version and license(s).

I am looking for an output in any format that would list this information for each dependency embedded in the final binary (no tests requirements).

Looking at the [envoy](https://github.com/envoyproxy/envoy) repo, it seems all this information is available to the build system: [envoy/bazel/repository_locations.bzl](https://github.com/envoyproxy/envoy/blob/main/bazel/repository_locations.bzl)

I have a `bazel` build command used to produce this `envoy` static binary with many 'filters' and 'options':
```
bazel \
  --output_base=output_base \
  build \
  --define google_grpc=disabled  \
  --@envoy//bazel:http3=False     \
  --@envoy//source/extensions/wasm_runtime/v8:enabled=false \
  [...]
  @envoy//source/exe:envoy-static
```

I have tried to play with the `query` command of `bazel` but I am not able to get what I want using it.

Do you know a way to get the output I need ?

Thank you.

**Comment by tjgq** [comment]
cc @aiuto

**Comment by aiuto** [comment]
we're working on various solutions for that with code in rules_license. But we're still working on the glue to help link back to the licenses of projects that come in auto-magically through maven deps and bzlmod.
